framework/api_vulkan_sample.h

```diff
@@ -99,7 +99,7 @@ struct Meshlet
  *
  * See vkb::VulkanSample for documentation
  */
-class ApiVulkanSample : public vkb::VulkanSample
+class ApiVulkanSample : public vkb::VulkanSample<false>
```

Comment:
Mostly out of curiosity, but what toes `<false>` mean in this regard and why is it needed now? Can't this be a default value that only needs to specified when ddifferent?

Reply:
The base class `VulkanSample` uses a boolean template parameter to determine whether to use the C++ or C types for various members, using `std::conditional` inside `VulkanSample`.

Reply:
As @jherico said.
One could make, for example, the C-binding type the default, reducing `VulkanSample<false>` to `VulkanSample<>` at lots of places. I don't think, that that would give you much.
I could imagine, though, to introduce some `enum class BindingType { CBinding, CppBinding };`, or so, and use that instead of just a bool. Would result in `VulkanSample<BindingType::CBinding>` and `VulkanSample<BindingType::CppBinding>`, which arguably is a bit more descriptive.